pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -244,11 +245,31 @@ func (c *Controller) initializeOperation(ctx context.Context, logger *logrus.Ent
 		return nil, fmt.Errorf("failed to get garden client: %w", err)
 	}
 
+	gardenSecrets, err := garden.ReadGardenSecrets(ctx, gardenClient.Cache(), c.seedLister, seedpkg.ComputeGardenNamespace(seed.Name))
+	if err != nil {
+		return nil, err
+	}
+
+	shootList, err := c.k8sGardenCoreInformers.Core().V1beta1().Shoots().Lister().List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	shootCount := len(shootList)
+	if len(shoot.Status.UID) == 0 {
+		// Do not consider shoot that is just being created.
+		shootCount -= 1
+	}
+
+	if err := garden.VerifyInternalDomainSecret(ctx, gardenClient, shootCount, gardenSecrets[common.GardenRoleInternalDomain]); err != nil {
+		return nil, err
+	}
```

Comment:
I guess this might be quite expensive and we should avoid it. I think we should rather remove this "verification" here in the code to the GAC and a validating webhook for this internal domain secret. For now, looking at the implementation of `VerifyInternalDomainSecret`, I guess it's sufficient to set `1` instead of the real shoot count (or `0` if the UID is empty).

Reply:
Makes sense, thank you 👍

Reply:
I had a second thought about tweaking this calculation but eventually noticed that this call shouldn't be expensive at all.
We use a lister, so the objects are anyway cached, and the lister even returns the original objects from the cache (no deep copy).

Reply:
Yeah, but we can probably still simplify this and save some code here.

Reply:
Do you have an idea how? As explained in my first comment, setting the count to `1` doesn't help.

Reply:
Sorry, I missed that. Yeah, you're right... OK, then let's keep it like this and rather move this code to GAC as planned. 👍🏻

Reply:
FYI: I have opened https://github.com/gardener/gardener/pull/3756 which implements the internal domain validation in the GAC as new webhook handler.